Source/WebKit/Platform/IPC/Decoder.h

```diff
@@ -129,7 +129,7 @@ class Decoder {
     template<typename T>
     std::optional<T> decode()
     {
-        std::optional<T> t { ArgumentCoder<std::remove_cvref_t<T>, void>::decode(*this) };
+        std::optional<T> t { ArgumentCoder<std::remove_cvref_t<T>>::decode(*this) };
```

Comment:
I'm surprised this is needed. While working on https://github.com/WebKit/WebKit/pull/27313 one of the WIP versions of the patch added a new `.serialization.in` and this did not need to be changed 🤔

Reply:
The signature of the encode methods autogenerated change a bit:

```
In file included from sources/wpewebkit/Source/WebKit/Shared/WebPreferencesStore.h:29,
                 from sources/wpewebkit/Source/WebKit/UIProcess/WebPreferences.h:31,
                 from wpewebkit-2.45.upstream-20240511/DerivedSources/WebKit/WebPreferencesFeatures.cpp:29:
sources/wpewebkit/Source/WebKit/Platform/IPC/Encoder.h: In member function 'IPC::Encoder& IPC::Encoder::operator<<(T&&)':
sources/wpewebkit/Source/WebKit/Platform/IPC/Encoder.h:80:46: error: wrong number of template arguments (2, should be 1)
   80 |         ArgumentCoder<std::remove_cvref_t<T, void>>::encode(*this, std::forward<T>(t));
      |                                              ^~~~
In file included from recipe-sysroot/usr/include/c++/13.2.0/bits/stl_pair.h:60,
                 from recipe-sysroot/usr/include/c++/13.2.0/bits/stl_algobase.h:64,
                 from recipe-sysroot/usr/include/c++/13.2.0/algorithm:60,
                 from sources/wpewebkit/Source/WebKit/WebKit2Prefix.h:60,
                 from <command-line>:
recipe-sysroot/usr/include/c++/13.2.0/type_traits:3442:11: note: provided for 'template<class _Tp> using std::remove_cvref_t = typename std::remove_cvref::type'
 3442 |     using remove_cvref_t = typename remove_cvref<_Tp>::type;
      |           ^~~~~~~~~~~~~~
sources/wpewebkit/Source/WebKit/Platform/IPC/Encoder.h:80:50: error: template argument 1 is invalid
   80 |         ArgumentCoder<std::remove_cvref_t<T, void>>::encode(*this, std::forward<T>(t));
      |                                                  ^~
sources/wpewebkit/Source/WebKit/Platform/IPC/Encoder.h:80:86: error: expression list treated as compound expression in initializer [-fpermissive]
   80 |         ArgumentCoder<std::remove_cvref_t<T, void>>::encode(*this, std::forward<T>(t));
      |                                                                                      ^
```

Reply:
However it seems this is an issue for the Apple ports:

```
In file included from /Volumes/Data/worker/macOS-Ventura-Release-Build-EWS/build/WebKitBuild/Release/DerivedSources/TestWebKitAPI/unified-sources/UnifiedSource40-mm.mm:3:
In file included from /Volumes/Data/worker/macOS-Ventura-Release-Build-EWS/build/Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:29:
/Volumes/Data/worker/macOS-Ventura-Release-Build-EWS/build/Source/WebKit/Platform/IPC/Encoder.h:80:9: error: too few template arguments for class template 'ArgumentCoder'
```

Reply:
I don't think you should do this change.